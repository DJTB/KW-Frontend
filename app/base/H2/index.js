import styled from 'styled-components';
import { beta, headingRhythm } from 'shared/styles/typography';

const H2 = styled.h2`
  ${beta}
  ${headingRhythm}
`;

export default H2;
