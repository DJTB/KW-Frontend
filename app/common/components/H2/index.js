import styled from 'styled-components';
import { beta, headingRhythm } from 'common/styles/typography';

const H2 = styled.h2`
  ${beta}
  ${headingRhythm}
`;

export default H2;
