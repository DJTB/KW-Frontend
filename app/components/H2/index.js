import styled from 'styled-components';
import { fluidType } from 'shared/styles/utils';
import { minFontSize, mobileMod, desktopMod } from 'shared/styles/sizing';

// Modular scale
const mobileSize = +((mobileMod ** 2.5) * minFontSize).toPrecision(3);
const desktopSize = +((desktopMod ** 2.5) * minFontSize).toPrecision(3);

const H2 = styled.h2`
  ${fluidType(mobileSize, desktopSize)}
`;

export default H2;
