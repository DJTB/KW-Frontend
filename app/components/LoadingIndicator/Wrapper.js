import styled from 'styled-components';

import { gutter } from 'shared/styles/layout';

const Wrapper = styled.div`
  ${gutter({ type: 'outer', prop: 'margin', direction: 'vertical' })}
  position: relative;
  width: 2rem;
  height: 2rem;
  margin-left: auto;
  margin-right: auto;
  color: ${({ color }) => color}
`;

export default Wrapper;
