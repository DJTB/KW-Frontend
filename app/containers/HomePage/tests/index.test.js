/**
 * Test the HomePage
 */


import { shallow, mount } from 'enzyme';
import React from 'react';

import { HomePage } from '../index';
import H2 from 'components/H2';
import LoadingIndicator from 'components/LoadingIndicator';

describe('<HomePage />', () => {
  it('should render the loading indicator when its loading', () => {
    const renderedComponent = shallow(
      <HomePage loading />
    );
    expect(renderedComponent.contains(<LoadingIndicator />).toBe(true));
  });

  it('should render an error if loading failed', () => {
    const renderedComponent = mount(
      <HomePage
        loading={false}
        error={{ message: 'Loading failed!' }}
      />
    );
    expect(
      renderedComponent
        .text()
        .indexOf('Something went wrong, please try again!')
      ).toBeGreaterThan(-1);
  });

  it('should render the userData if loading was successful', () => {
    const user = {
      name: 'testname',
      level: 3,
      reviewCount: 2,
      lastWkSyncDate: new Date(),
    };
    const renderedComponent = shallow(
      <HomePage
        user={user}
        error={false}
      />
    );

    expect(renderedComponent.contains(
      <div>
        <H2>Welcome Back testname.</H2>
        <p>You are level 3.</p>
        <p>You have 2 reviews waiting.</p>
        <p>You last synced with WK on {new Date().toDateString()}.</p>
      </div>
    )).toBe(true);
  });

 /* describe('mapDispatchToProps', () => {
    describe('onChangeUsername', () => {
      it('should be injected', () => {
        const dispatch = expect.createSpy();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeUsername).toExist();
      });

      it('should dispatch changeUsername when called', () => {
        const dispatch = expect.createSpy();
        const result = mapDispatchToProps(dispatch);
        const username = 'mxstbr';
        result.onChangeUsername({ target: { value: username } });
        expect(dispatch).toHaveBeenCalledWith(changeUsername(username));
      });
    });
  });

  describe('onSubmitForm', () => {
    it('should be injected', () => {
      const dispatch = expect.createSpy();
      const result = mapDispatchToProps(dispatch);
      expect(result.onSubmitForm).toExist();
    });

    it('should dispatch loadUserData when called', () => {
      const dispatch = expect.createSpy();
      const result = mapDispatchToProps(dispatch);
      result.onSubmitForm();
      expect(dispatch).toHaveBeenCalledWith(loadUserData());
    });

    it('should preventDefault if called with event', () => {
      const preventDefault = expect.createSpy();
      const result = mapDispatchToProps(() => {});
      const evt = { preventDefault };
      result.onSubmitForm(evt);
      expect(preventDefault).toHaveBeenCalledWith();
    });
  });*/
});
