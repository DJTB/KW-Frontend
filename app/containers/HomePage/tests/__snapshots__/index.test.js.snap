// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match the baseline snapshot 1`] = `
<h1>
  Hello HomePage
  <label
    htmlFor="derp"
  >
    toggle me
    <ToggleSwitch
      handleChange={[Function]}
      height="2rem"
      id="derp"
      isChecked={false}
      name="derp"
      toggleOffColor="red"
      toggleOnColor="green"
      width="6rem"
    />
  </label>
</h1>
`;
